### PR TITLE
ci: Clean up logic for `merge -r`

### PR DESCRIPTION
**Pull request by seemethere**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #114295

Rely on built in bash conditionals for doing the if statement rather
than relying on $?

To avoid issues observed in https://github.com/pytorch/pytorch/pull/111008#issuecomment-1821547141

Signed-off-by: Eli Uriegas <eliuriegas@meta.com>

### CODE REVIEW

**Review by huydhn — APPROVED** [review]
LGTM!

**Comment by seemethere** [comment]
I also don't know if there's a good way to test this without a YOLO

File: .github/workflows/trymerge.yml [hunk 1]
```diff
@@ -45,8 +45,8 @@ jobs:
         run: |
```
Comment (malfet) [inline .github/workflows/trymerge.yml:45]:
```suggestion
        shell: bash
        run: |
```

**Review by malfet — APPROVED** [review]
LGTM, please also add `shell: bash` to avoid any unexpected changes

File: .github/workflows/trymerge.yml (hunk 1 above)
Comment (seemethere) [reply .github/workflows/trymerge.yml:46]: Added above

**Pull request closed without merging**